src/Storefront/Framework/Routing/Router.php

```diff
@@ -168,7 +168,7 @@ private function removePrefix(string $subject, string $prefix): string
 
     private function getSalesChannelBaseUrl(): string
     {
-        $request = $this->requestStack->getMainRequest();
```

Comment:
Anyone has an idea whether this may break some scenarios and you actually want to generate the URL based on the main request? 🤔 

I couldn't come up with any, and the tests also don't cover any, so I would think it is safe to make the change 

I solved the same issue in a more hacky way already here: https://github.com/shopware/shopware/pull/9159

But didn't merge because we were not sure what the root issue actually was, that would an alternative solution, when we think changing the router logic here is too risky 

Wanting to hear your opinion @shyim & @mitelg